discovery-provider/src/tasks/playlists.py

```diff
@@ -25,67 +27,78 @@ def playlist_state_update(
     block_hash,
 ):
     """Return int representing number of Playlist model state changes found in transaction."""
+    blockhash = update_task.web3.toHex(block_hash)
     num_total_changes = 0
+    skipped_tx_count = 0
     # This stores the playlist_ids created or updated in the set of transactions
     playlist_ids = set()
 
     if not playlist_factory_txs:
         return num_total_changes, playlist_ids
 
-    playlist_abi = update_task.abi_values["PlaylistFactory"]["abi"]
-    playlist_contract = update_task.web3.eth.contract(
-        address=get_contract_addresses()["playlist_factory"], abi=playlist_abi
-    )
-
     playlist_events_lookup = {}
     for tx_receipt in playlist_factory_txs:
         txhash = update_task.web3.toHex(tx_receipt.transactionHash)
         for event_type in playlist_event_types_arr:
-            playlist_events_tx = getattr(
-                playlist_contract.events, event_type
-            )().processReceipt(tx_receipt)
+            playlist_events_tx = get_playlist_events_tx(
+                update_task, event_type, tx_receipt
+            )
             processedEntries = 0  # if record does not get added, do not count towards num_total_changes
             for entry in playlist_events_tx:
                 try:
-                    playlist_id = entry["args"]._playlistId
-                    playlist_ids.add(playlist_id)
+                    playlist_id = get_tx_arg(entry, "_playlistId")
 
                     if playlist_id not in playlist_events_lookup:
                         existing_playlist_entry = lookup_playlist_record(
                             update_task, session, entry, block_number, txhash
                         )
-                        playlist_events_lookup[playlist_id] = {
-                            "playlist": existing_playlist_entry,
-                            "events": [],
-                        }
-
-                    playlist_record = parse_playlist_event(
-                        self,
-                        update_task,
-                        entry,
-                        event_type,
-                        playlist_events_lookup[playlist_id]["playlist"],
-                        block_timestamp,
-                        session,
-                    )
+                    try:
+                        playlist_record = parse_playlist_event(
+                            self,
+                            update_task,
+                            entry,
+                            event_type,
+                            existing_playlist_entry,
```

Comment:
this scoping looks a bit confusing because existing_playlist_entry is declared inside the if and this is called inside the try. they're both function scoped but because of indenting it's hard to follow